docs/DeveloperGuide.md

```diff
@@ -720,7 +832,88 @@ testers are expected to do more *exploratory* testing.
 
 [trackAScholarImport test files](https://github.com/AY2223S1-CS2103T-W10-3/tp/tree/master/src/test/data/JsonImportCommandTest)
 
+--------------------------------------------------------------------------------------------------------------------
+
+### Pinning an applicant
+
+1. Pinning an existing applicant while all applicants are being shown.
+
+    1. Prerequisites: List all applicants using the `list` command. Multiple applicants in the list. 
 
+    1. Test case: `pin 1`<br>
+       Expected: First applicant in the list is pinned. Details of the pinned applicant shown in the status message. 
+       Pinned Applicant will be shown on the right list panel. Timestamp in the status bar is updated.
+
+    1. Test case: `pin 0`<br>
+       Expected: No applicant is pinned. Error details shown in the status message. Status bar remains the same.
+
+    1. Other incorrect pin commands to try: `pin`, `pin x`, `...` (where x is larger than the list size)<br>
+       Expected: Similar to previous.
+
+--------------------------------------------------------------------------------------------------------------------
+
+### Unpinning an applicant
+
+1. Unpinning an existing applicant while all applicants are being shown
+
+    1. Prerequisites: List all applicants using the `list` command. One applicant in the list. The list contains
+       a pinned applicant named `Alex Yeoh`.
+
+    1. Test case: `unpin Alex Yeoh`<br>
+       Expected: Existing applicant with name `Alex Yeoh` in the list is unpinned. Applicant will be removed from
+       the right list panel. Details of the unpinned applicant shown in the status message. Timestamp in the status bar is updated.
+
+    1. Test case: `unpin Benjamin Loy`<br>
+       Expected: No changes displayed on TrackAScholar GUI as the list does not contain an applicant with name `Benjamin Loy`.
+       Error details shown in the status message. Status bar remains the same.
+
+    1. Other incorrect unpin commands to try: `unpin` and `unpin 123` <br>
```

Comment:
Perhaps consider adding the test case where the user attempts to unpin an already unpinned applicant?